nautobot/ipam/forms.py

```diff
@@ -776,6 +788,14 @@ def __init__(self, *args, **kwargs):
         else:
             self.fields["ip_addresses"].choices = []
 
+    def clean(self, *args, **kwargs):
+        if self.instance:
+            if self.instance.device:
+                self.cleaned_data["device"] = self.instance.device
+            else:
+                self.cleaned_data["virtual_machine"] = self.instance.virtual_machine
```

Comment:
What's this needed for?

Reply:
I think because I added `device` and `virtual_machine` fields on `ServiceForm` and in `service_edit.html`, we only statically render `obj.device` or `obj.virtual_machine` in EditView. Because of that, self.cleaned_data["device"] and self.cleaned_data["virtual_machine"] will always be `None`, this will cause a `ValidationError` in `clean()`.

Reply:
On second thought, can `Device/VirtualMachine` be modified in `ServiceEditView`? If so, we can always render the fields dynamically.